.github/workflows/workflow-cleanup.yml

```
@@ -0,0 +1,47 @@
+## Github workflow to clean up old caches and workflow runs
+
+name: Workflow Cleanup
+
+on:
+  workflow_dispatch:
+    inputs:
+      cache-ttl:
+        description: "How many days to keep a cache (default: 7)"
+        required: false
+        default: "7"
+      workflow-ttl:
+        description: "How many days to keep a successful workflow (default: 30)"
+        required: false
+        default: "30"
+      failed-workflow-ttl:
+        description: "How many days to keep failed workflows (default: 15)"
+        required: false
+        default: "15"
+  schedule:
+    ## Run every day at 00:00:00
+    - cron: "0 0 * * *"
+
+## env vars are transferred to composite action steps
+env:
+  CACHE_TTL: 7 ## number of days to keep a cache
+  WORKFLOW_TTL: 30 ## number of days to keep a successful workflow
+  FAILED_WORKFLOW_TTL: 15 ## number of days to keep a failed workflow
+
+concurrency:
+  group: cleanup-${{ github.head_ref || github.ref }}
```

Is this intended to be:
```suggestion
  group: cleanup-${{ github.head_ref || github.ref || github.run_id}}
```
 following the concurrency pattern set in your other workflows?